torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -773,6 +773,15 @@ bool LoopNest::computeInline(Stmt* s) {
 }
 
 bool LoopNest::computeInline(const Buf* b) {
+  // If buf is used or defined in an ExternalCall, we cannot inline it
```

Comment:
Why shouldn't we inline if a buf is used in an ExternalCall? IIUC, for such cases, we should be able to inline  but still leave the original computation as is, so that it can be used in the ExternalCall.

Am I missing something? Or Does inlining not support this feature?

Reply:
External calls take Bufs as arguments, not an expression for it, so there is nothing we could inline to.

Another way to look at it is the following. Imagine you have a following C++ program:
```
for (int i = 0; i < 100; i++)
  a[i] = i*i;
for (int j = 0; j < 100; j++)
  b[j] = a[j] + 2;
c = at::some_op(b);
```
Inlining `a` to `b` makes perfect sense: we just replace `a[j]` with `j*j`. Inlining `b` to `c` doesn't really make sense - what could we do there?

Reply:
Not inlining a buffer at the use-point in an ExternalCall makes sense. But what if the same buffer is used in another computation as well? 

IIUC, any buffer that is used even once in an ExternalCall will not be inlined here, right? My point is, if there are other uses of such buffers, they should be inlined.

Reply:
Ah, I see your point! Yes, you are right! Do you mind if I do that in a follow-up? I think it would be slightly easier to pull off once we get rid of intermediates_bufs_ in loopnest which I hope to get to very soon.

Reply:
Sure, sounds good.